### PR TITLE
Task ys/j108 add my podcasts page

### PR DESCRIPTION
**Pull request by yaroslav207**

### CODE REVIEW

File: shared/src/common/enums/api/podcasts-api-path.enum.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 enum PodcastsApiPath {
   ROOT = '/',
   $ID = '/:id',
+  USER = '/user',
+  USER_$ID = '/user/:id',
```
Comment (ghost) [inline shared/src/common/enums/api/podcasts-api-path.enum.ts]:
```suggestion
  USERS = '/users',
  USERS_$ID = '/users/:id',
```

File: frontend/src/services/podcast-api/podcast-api.service.ts [hunk 2]
```diff
@@ -60,6 +60,16 @@ class PodcastApi {
       },
     );
   }
+
+  public getPodcastsByUserId(userId: number): Promise<Podcast[]> {
```
Comment (ghost) [inline frontend/src/services/podcast-api/podcast-api.service.ts]:
```suggestion
  public getAllByUserId(userId: number): Promise<Podcast[]> {
```

File: backend/src/services/podcast/podcast.service.ts [hunk 3]
```diff
@@ -124,6 +124,10 @@ class Podcast {
 
     return podcast;
   }
+
+  public getPodcastsByUserId(userId: string): Promise<TPodcast[]> {
+    return this.#podcastRepository.getPodcastsByUserId(userId);
```
Comment (ghost) [inline backend/src/services/podcast/podcast.service.ts]:
```suggestion
  public getAllByUserId(userId: string): Promise<TPodcast[]> {
    return this.#podcastRepository.getAllByUserId(userId);
```

File: backend/src/data/repositories/podcast/podcast.repository.ts [hunk 4]
```diff
@@ -32,6 +32,10 @@ class Podcast {
   public update(id: string, payload: PodcastEditDTOPayload): Promise<TPodcast> {
     return this.#PodcastModel.query().updateAndFetchById(id, payload);
   }
+
+  public getPodcastsByUserId(userId: string): Promise<TPodcast[]> {
```
Comment (ghost) [inline backend/src/data/repositories/podcast/podcast.repository.ts]:
```suggestion
  public getAllByUserId(userId: string): Promise<TPodcast[]> {
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by yevhenprybytok — DISMISSED** [review]

File: shared/src/common/enums/api/podcasts-api-path.enum.ts [hunk 5]
```diff
@@ -1,6 +1,8 @@
 enum PodcastsApiPath {
   ROOT = '/',
   $ID = '/:id',
+  USER = '/users',
+  USER_$ID = '/users/:id',
```
Comment (ghost) [inline shared/src/common/enums/api/podcasts-api-path.enum.ts]:
```suggestion
  USERS = '/users',
  USERS_$ID = '/users/:id',
```

File: frontend/src/services/podcast-api/podcast-api.service.ts [hunk 6]
```diff
@@ -60,6 +60,16 @@ class PodcastApi {
       },
     );
   }
+
+  public getAllByUserId(userId: number): Promise<Podcast[]> {
+    return this.#http.load(
+      `${this.#apiPrefix}${ApiPath.PODCASTS}${PodcastsApiPath.USER}${userId}`,
```
Comment (ghost) [inline frontend/src/services/podcast-api/podcast-api.service.ts]:
```suggestion
      `${this.#apiPrefix}${ApiPath.PODCASTS}${PodcastsApiPath.USER}/${userId}`,
```
🤔

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — CHANGES_REQUESTED** [review]
add output podcast functionality for user-profile page

File: frontend/src/components/user-profile/user-profile.tsx [hunk 7]
```diff
@@ -73,6 +84,16 @@ const UserPage: React.FC = () => {
       <div className={styles.favoritePodcastContainer}>
         <h2 className={styles.favoritePodcastTitle}>Favorite Podcasts</h2>
         <div className={styles.podcasts}></div>
+        <h2 className={styles.favoritePodcastTitle}>My Podcasts</h2>
+        {dataStatus === DataStatus.PENDING
+          ? <Loader />
+          : podcasts ? (
```
Comment (ghost) [inline frontend/src/components/user-profile/user-profile.tsx]: podcasts is an array. array is always true

File: frontend/src/components/user-profile/user-profile.tsx [hunk 8]
```diff
@@ -1,16 +1,23 @@
...
 import editLogo from 'assets/img/user-profile/edit.svg';
 import emailLogo from 'assets/img/user-profile/email.svg';
 import defaultImage from 'assets/img/user-profile/default-profile-picture.jpg';
 import styles from './styles.module.scss';
+import { userProfile as userProfileActions } from 'store/actions';
+import { DataStatus } from 'common/enums/enums';
 
 const UserPage: React.FC = () => {
-  const { user } = useAppSelector(({ auth }: RootState) => ({
+  const { user, podcasts, dataStatus } = useAppSelector(({ auth, userProfile }: RootState) => ({
```
Comment (ghost) [inline frontend/src/components/user-profile/user-profile.tsx]: you need to fetch user for this page

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/components/user-profile/user-profile.tsx (hunk 8 above)
Comment (yaroslav207) [reply frontend/src/components/user-profile/user-profile.tsx]: > you need to fetch user for this page

I didn't quite understand.
We get user from the store.

Comment (ghost) [reply frontend/src/components/user-profile/user-profile.tsx]: you need to create user service, reducer and etc 🙂
this page will be public

File: backend/src/api/users/users.api.ts [hunk 9]
```diff
@@ -22,6 +22,14 @@ const initUsersApi = ({ apiRouter, userService }: Args): Router => {
     }),
   );
 
+  userRouter.get(
+    UsersApiPath.$ID,
+    checkAuth(HttpMethod.GET),
+    handleAsyncApi(async (_req, res) => {
```
Comment (ghost) [inline backend/src/api/users/users.api.ts:28]:
```suggestion
    handleAsyncApi(async (req, res) => {
```
underscore only for unused params

**Review by ghost — APPROVED** [review]

**Pull request merged**